engine/compiler/compiler.go

```diff
@@ -589,52 +589,69 @@ func (c *Compiler) Compile(ctx context.Context, args runtime.CompilerArgs) runti
 		spec.Volumes = append(spec.Volumes, src)
 	}
 
-	// apply default resources limits
-	for _, v := range spec.Steps {
-		if v.Resources.Limits.CPU == 0 {
-			v.Resources.Limits.CPU = c.Resources.Limits.CPU
-		}
-		if v.Resources.Limits.Memory == 0 {
-			v.Resources.Limits.Memory = c.Resources.Limits.Memory
-		}
+	// apply policy
```

Comment:
it might be worth moving some of the excellent comments above the m.apply(spec) call.

Reply:
well, I'd rather not because policy is not directly linked to resources distribution. policy just overrides pipeline configuration...

Reply:
even that line would be good. `// policies overrides pipeline configuration`